### PR TITLE
Support lazily created SDK configurations.

### PR DESCRIPTION
**Pull request by annagrin**
- Add SdkConfigurationInterface and a default implementation.
- Pass SdkConfiguration to dwds and ExpressionCompilerService.
- Add tests.

Note: this change will enable us to add debugging for scenarios where SDK files are generated and copied during the build.

### CODE REVIEW

File: dwds/lib/src/utilities/sdk_configuration.dart [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  Future<String> get sdkDirectory;
+  Future<String> get unsoundSdkSummaryPath;
+  Future<String> get soundSdkSummaryPath;
+  Future<String> get librariesPath;
+  Future<String> get compilerWorkerPath;
+
+  Future<Uri> get sdkDirectoryUri =>
+      sdkDirectory.then((value) => value == null ? null : p.toUri(value));
+
+  Future<Uri> get soundSdkSummaryUri => soundSdkSummaryPath
```
Comment (grouma) [inline dwds/lib/src/utilities/sdk_configuration.dart:39]: Curious why these have to be async? Would it be better to have an async "constructor"? Or would it be slightly cleaner to have an async init function?

Comment (grouma) [reply dwds/lib/src/utilities/sdk_configuration.dart:39]: I think your suggestion seems better as it makes it clear why we need `async`.

File: dwds/lib/src/services/chrome_proxy_service.dart [hunk 2]
```diff
@@ -1035,6 +1041,10 @@ ${globalLoadStrategy.loadModuleSnippet}("dart_sdk").developer.invokeExtension(
           String isolateId, String breakpointId, bool enable) =>
       throw UnimplementedError();
 
+  @override
```
Comment (elliette) [inline dwds/lib/src/services/chrome_proxy_service.dart:1044]: Is this intended to be included in this change?

Comment (annagrin) [reply dwds/lib/src/services/chrome_proxy_service.dart:1045]: I needed it to test that everything works. Apparently the other PR does not fix the bug it intended to, so I might keep it here instead and abandon the other.

**Review by grouma — APPROVED** [review]

File: dwds/lib/src/services/chrome_proxy_service.dart (hunk 2 above)
Comment (annagrin) [reply dwds/lib/src/services/chrome_proxy_service.dart:1045]: Let me know if you prefer it as a separate change.

**Pull request merged**